tools/omnissm/pkg/manager/manager.go

```diff
@@ -15,39 +15,68 @@
 package manager
 
 import (
+	"errors"
+	"time"
+
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ssm"
 	"github.com/aws/aws-sdk-go/service/ssm/ssmiface"
+	"github.com/jpillora/backoff"
 
 	"github.com/capitalone/cloud-custodian/tools/omnissm/pkg/identity"
 	"github.com/capitalone/cloud-custodian/tools/omnissm/pkg/store"
 )
 
-const DefaultSSMServiceRole = "service-role/AmazonEC2RunCommandRoleForManagedInstances"
+const (
+	DefaultMaxRetries     = 5
+	DefaultSSMServiceRole = "service-role/AmazonEC2RunCommandRoleForManagedInstances"
+)
+
+var ErrMaxRetriesExceeded = errors.New("max retries exceeded")
+
+func retry(maxRetries int, fn func() error) error {
+	b := &backoff.Backoff{Min: 1 * time.Second, Max: 30 * time.Second, Factor: 2}
+	for int(b.Attempt()) < maxRetries {
+		if err := fn(); err != nil {
+			if request.IsErrorThrottle(err) || request.IsErrorRetryable(err) {
+				time.Sleep(b.Duration())
```

Comment:
can we log somewhere in the retry for debug purposes.

Reply:
i believe the error can be wrapped before return here and provide the context and the desired utility, thoughts?